scripts/cmake/z_vcpkg_catch_ambiguous_system_variables.cmake

```
@@ -0,0 +1,17 @@
+function(z_vcpkg_catch_ambiguous_system_variables VARIABLE ACCESS VALUE POS STACK)
+    message(FATAL_ERROR "Unexpected ${ACCESS} on variable ${VARIABLE} in script mode.
+This variable name insufficiently expresses whether it refers to the \
+target system or to the host system. Use a prefixed variable instead.
+- Variables providing information about the host:
+  CMAKE_HOST_<SYSTEM>
+  VCPKG_HOST_IS_<SYSTEM>
+- Variables providing information about the target:
+  VCPKG_TARGET_IS_<SYSTEM>
+  VCPKG_DETECTED_<VARIABLE> (using vcpkg_cmake_get_vars)
+")
+endfunction()
+
+set(VCPKG_AMBIGUOUS_SYSTEM_VARIABLES "" CACHE STRING "Variables which must not be used in portfiles")
+foreach(var IN LISTS VCPKG_AMBIGUOUS_SYSTEM_VARIABLES)
+    variable_watch("${var}" z_vcpkg_catch_ambiguous_system_variables)
+endforeach()
```

```suggestion
function(z_vcpkg_catch_ambiguous_system_variables VARIABLE ACCESS VALUE POS STACK)
    message("${Z_VCPKG_BACKCOMPAT_MESSAGE_LEVEL}" "Unexpected ${ACCESS} on variable ${VARIABLE} in script mode.
This variable name insufficiently expresses whether it refers to the \
target system or to the host system. Use a prefixed variable instead.
- Variables providing information about the host:
  CMAKE_HOST_<SYSTEM>
  VCPKG_HOST_IS_<SYSTEM>
- Variables providing information about the target:
  VCPKG_TARGET_IS_<SYSTEM>
  VCPKG_DETECTED_<VARIABLE> (using vcpkg_cmake_get_vars)
")
endfunction()

foreach(var IN ITEMS ANDROID APPLE BSD IOS LINUX MINGW MSVC UNIX WIN32)
    variable_watch("${var}" z_vcpkg_catch_ambiguous_system_variables)
endforeach()
```

Also I think we should just expand this file inline in `ports.cmake` rather than needing to open another file.